media-proxy/src/libfabric_cq.c

```diff
@@ -78,43 +81,31 @@ void rdma_cq_set_wait_attr(struct fi_cq_attr *cq_attr, enum cq_comp_method metho
     }
 }
 
-static int rdma_spin_for_comp(ep_ctx_t *ep_ctx, struct fid_cq *cq, uint64_t *cur, uint64_t total,
-                              int timeout, struct fi_cq_err_entry *entries)
+static int rdma_spin_for_comp(struct ep_ctx_t *ep_ctx, struct fi_cq_err_entry *entry, int timeout)
 {
-    uint64_t entries_num;
-    struct fi_cq_err_entry comp;
-    struct timespec a, b;
+    struct timespec start_time, current_time;
     int ret;
 
-    if (!cur)
-        return -EINVAL;
-
-    if (total < *cur)
-        return -EINVAL;
-
-    entries_num = total - *cur;
-
     if (timeout >= 0)
-        clock_gettime(CLOCK_MONOTONIC, &a);
+        clock_gettime(CLOCK_MONOTONIC, &start_time);
 
-    do {
-        ret = fi_cq_read(cq, entries ? &entries[entries_num - (total - *cur)] : &comp, 1);
+    while (1) {
+        ret = fi_cq_read(ep_ctx->cq_ctx.cq, entry, 1);
         if (ret > 0) {
-            if (timeout >= 0)
-                clock_gettime(CLOCK_MONOTONIC, &a);
-            (*cur)++;
+            ep_ctx->cq_ctx.cq_cntr++;
+            return 0;
         } else if (ret < 0 && ret != -FI_EAGAIN) {
             return ret;
-        } else if (timeout >= 0) {
-            clock_gettime(CLOCK_MONOTONIC, &b);
-            if ((b.tv_sec - a.tv_sec) > timeout) {
+        }
+
+        if (timeout >= 0) {
```

Comment:
This code will be running when `ret == 0` or `ret == -FI_EAGAIN`, correct?
If yes, please add a comment explaining that.

Reply:
Correct. ret is of the ssizze_t type, it returns number of cq entries returned or an error.
So we are satisfied when ret == 1,  as we requesting just one entry, then we want to return 0, means success.
if any error other then -FI_EAGAIN occurred we want to immediately return it.
-FI_EAGAIN means that nothing is in the queue so we're spin waiting for something in queue ( and are checking if no time out).

I'm not sure if ret ever will be equal to 0, but anyway it does not mean success.

Reply:
Having such a comment in the code would be very helpful for future efforts of development.

Reply:
Do you think a comment in code is necessary? maybe I will change ret to ssizze_t  to better communicate  intention

Reply:
comments added :)